packages/appium/lib/config.js

```diff
@@ -44,11 +44,23 @@ async function updateBuildInfo (useGithubApiFallback = false) {
   }
 }
 
+/**
+ * Finds the Git metadata dir (see `GIT_META_ROOT`)
+ *
+ * This is needed because Appium cannot assume `package.json` and `.git` are in the same
+ * directory.  Monorepos, see?
+ * @returns {string|void} Path to dir or `undefined` if not found
+ */
+async function findGitRoot () {
+  return await findUp(GIT_META_ROOT, {cwd: rootDir, type: 'directory'});
```

Comment:
cannot we use the existing glob package from appium-support for this purpose instead of adding a new dependency?

Reply:
I mean, we could, but we'd end up writing our own `find-up`.

Reply:
afaict there's no wildcard that tells glob/minimatch to navigate to an arbitrary parent directory from cwd.  you'd need to start from the fs root and work forwards, which would be...slow.  so you'd need to search for `.git` using glob, and if it's not there, navigate to `..`, then search again (but skip the directory we just came from), and do so in a loop.  I imagine this is basically what `find-up` is doing

Reply:
np, I'm just usually being a bit paranoidal about adding new dependecies because some of them might exponentially increase the size of the linked modules tree %)